lib/jellyfish/config_reader.ex

```diff
@@ -128,12 +103,121 @@ defmodule Jellyfish.ConfigReader do
     end
   end
 
-  defp parse_node_name(node_name), do: String.to_atom(node_name)
+  def read_dist_config() do
+    dist_enabled? = read_boolean("JF_DIST_ENABLED")
+    dist_strategy = System.get_env("JF_DIST_STRATEGY_NAME")
+    node_name_value = System.get_env("JF_DIST_NODE_NAME")
+    cookie_value = System.get_env("JF_DIST_COOKIE", "jellyfish_cookie")
+
+    cond do
+      is_nil(dist_enabled?) or not dist_enabled? ->
+        [enabled: false, strategy: nil, node_name: nil, cookie: nil, strategy_config: nil]
+
+      dist_strategy == "NODES_LIST" or is_nil(dist_strategy) ->
+        nodes_value = System.get_env("JF_DIST_NODES", "")
+
+        unless node_name_value do
+          raise "JF_DIST_ENABLED has been set but JF_DIST_NODE_NAME remains unset."
+        end
+
+        node_name = parse_node_name(node_name_value)
+        cookie = parse_cookie(cookie_value)
+        nodes = parse_nodes(nodes_value)
+
+        if nodes == [] do
+          Logger.warning("""
+          JF_DIST_ENABLED has been set but JF_DIST_NODES remains unset.
+          Make sure that at least one of your Jellyfish instances
+          has JF_DIST_NODES set.
+          """)
+        end
+
+        [
+          enabled: true,
+          strategy: Cluster.Strategy.Epmd,
+          node_name: node_name,
+          cookie: cookie,
+          strategy_config: [hosts: nodes]
+        ]
+
+      dist_strategy == "DNS" ->
+        do_read_dns_config(dist_strategy, node_name_value, cookie_value)
+
+      true ->
+        raise """
+        JF_DIST_ENABLED has been set but unknown JF_DIST_STRATEGY was provided.
+        Availabile strategies are EPMD or DNS, provided strategy name was: "#{dist_strategy}"
+        """
+    end
+  end
+
+  defp do_read_dns_config(dist_strategy, node_name_value, cookie_value) do
+    unless node_name_value do
+      raise "JF_DIST_ENABLED has been set but JF_DIST_NODE_NAME remains unset."
+    end
+
+    node_name = parse_node_name(node_name_value)
+    cookie = parse_cookie(cookie_value)
+
+    parse_dns_string = fn env_name ->
+      env = System.get_env(env_name)
+
+      if !env and dist_strategy == "DNS" do
+        raise "DNS strategy has been set but #{env_name} remains unset."
+      end
+
+      env
+    end
+
+    query = parse_dns_string.("JF_DIST_QUERY")
+
+    [node_basename, _ip_addres_or_fqdn | []] = String.split(node_name_value, "@")
```

Comment:
Leftover?

Reply:
Nope, libcluster requires to specify `node_basename` and here we parse `node_name_value` to extract it. Previously we read another environment variable.